### PR TITLE
Github Actions Build

### LINKED ISSUE #12: Create a GitHub Actions pipeline
The pipeline should build the code and run the tests. I recommend separating these two tasks into two different pipeline. 
Also, make sure to configure that pull requests can't be merged if the pipeline fails.

### PR DESCRIPTION
**Pull request by adamv17**
This pull request is intended to add github actions to protect the main branch and make sure that code in the main branch is buildable. closes #12

### CODE REVIEW

**Comment by adamv17** [comment]
Is there a way to have it be able to run not only on the latest version of windows, so that we don't have problems with this later?

**Comment by barel13** [comment]
I don't think so.
Also, I think you can use wpilib [example](https://docs.wpilib.org/en/stable/docs/software/advanced-gradlerio/robot-code-ci.html)
 for that kind of Action

File: .github/workflows/gradle.yml [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+# This workflow will build a Java project with Gradle and cache/restore any dependencies to improve the workflow execution time
+# For more information see: https://help.github.com/actions/language-and-framework-guides/building-and-testing-java-with-gradle
+
+name: Java CI with Gradle
+
+on:
+  push:
+    branches: [ main ]
+  pull_request:
+    branches: [ main ]
```
Comment (katzuv) [inline .github/workflows/gradle.yml]: Maybe trigger this action for every pull request, not just those opened against `main`.

**Comment by katzuv** [comment]
> Is there a way to have it be able to run not only on the latest version of windows, so that we don't have problems with this later?

Do you mean for Linux too? IMO it's not necessary, just run it on any OS and it should be fine (the code is the same, and if anything the roboRIO is Linux-based). 
(P.S: not sure how it would run on Windows, in Linux you need to grant `gradlew` executing permission (`chmod +x`), not sure if and how it's done on Linux).
I'm seconding @barel13's recommendation, Actions might be painful and it's better to use an existing one. You can also use the Action from 2020 and 2021 repositories.

File: .github/workflows/gradle.yml [hunk 2]
```diff
@@ -0,0 +1,30 @@
+# This is a basic workflow to build robot code.
+
+name: CI
+
+# Controls when the action will run. Triggers the workflow on push or pull request
+# events but only for the main branch.
```
Comment (katzuv) [inline .github/workflows/gradle.yml]: This comment lies.

**Review by katzuv — CHANGES_REQUESTED** [review]

**Review by barel13 — APPROVED** [review]

File: .github/workflows/gradle.yml (hunk 1 above)
Comment (adamv17) [reply .github/workflows/gradle.yml]: Done

File: .github/workflows/gradle.yml [hunk 3]
```diff
@@ -0,0 +1,34 @@
+# This is a basic workflow to build robot code.
+
+name: CI
+
+# Controls when the action will run. Triggers the workflow for all push or pull request
```
Comment (katzuv) [inline .github/workflows/gradle.yml]:
```suggestion
# Controls when the action will run. Triggers the workflow for all pushes or pull requests.
```

**Comment by katzuv** [comment]
Great! Just make sure you write your own commit messages, GitHub's default "Update ..." isn't meaningful.

**Review by katzuv — APPROVED** [review]

**Pull request merged**